Guarin lig 3049 add knn classifier tests

guarin opened the pull request:
## Changes

* Add tests for KNNClassifier

MalteEbner commented on tests/utils/benchmarking/test_knn_classifier.py:
```diff
@@ -0,0 +1,129 @@
+        model = nn.Identity()
+        classifier = KNNClassifier(model, num_classes=4, knn_k=3, topk=(1, 2, 3, 4))
+        trainer = Trainer(max_epochs=1, accelerator="cpu", devices=1)
+        train_features = torch.tensor(
+            [
+                [1.0, 1.0, 1.0, 1.0],
+                [-1.0, 1.0, 1.0, 1.0],
+                [-1.0, -1.0, 1.0, 1.0],
+                [-1.0, -1.0, -1.0, 1.0],
+            ]
+        )
+        train_features = F.normalize(train_features, dim=1)
+        train_targets = torch.tensor([0, 1, 2, 3])
+        train_dataset = _FeaturesDataset(features=train_features, targets=train_targets)
+        val_features = torch.tensor(
+            [
+                [1.0, 1.0, 1.0, 1.0],  # predicted_labels = [0, 1, 2, 3]
+                [-0.8, 1.0, 1.0, 1.0],  # predicted_labels = [1, 0, 2, 3]
+                [-1.2, 1.0, -1.0, 1.0],  # predicted_labels = [1, 3, 0, 2]
+            ]
+        )
```
Could you explain the ideas of this test in more detail?
Furthermore, I think that the features are unncecessarily complex, making it hard to understand. Neither 4 dimensions nor negative numbers are need.
Why not

```python
# define 4 training points from 4 classes 
train_features = torch.tensor(
    [
        [0.0, 0.0],
        [0.0, 1.0],
        [1.0, 0.0],
        [1.0, 1.0],
    ]
)
train_features = F.normalize(train_features, dim=1)
train_targets = torch.tensor([0, 1, 2, 3])
# define 3 validation points
# their expected predicted labels are their closest training points in order
val_features = torch.tensor(
    [
        [0.0, 0.4],  # predicted_labels = [0, 1, 2, 3]
        [0.6, 0.7],  # predicted_labels = [3, 1, 2, 0]
        [0.6, 0.3],  # predicted_labels = [2, 0, 3, 1]
    ]
)
```

MalteEbner reviewed (approved):
I think that the test could be made easier to understand, I spent quite a while on it.
Otherwise, this is fine, good work!

guarin commented:
> I think that the test could be made easier to understand, I spent quite a while on it.

Sorry for this, I updated the test.

I also removed feature normalization as it is handled by the classifier.
